utilities_common/netstat.py

```diff
@@ -56,6 +65,41 @@ def ns_util(newstr, oldstr, delta, port_rate=PORT_RATE):
         util = rate/(port_rate*1000*1000*1000/8.0)*100
         return "{:.2f}%".format(util)
 
+def format_brate(rate):
+    """
+    Show the byte rate.
+    """
+    if rate == STATUS_NA:
+        return STATUS_NA
+    else:
+        rate = float(rate)
+        if rate > 1024*1024*10:
+            rate = "{:.2f}".format(rate/1024/1024.0)+' MB'
+        elif rate > 1024*10:
+            rate = "{:.2f}".format(rate/1024.0)+' KB'
+        else:
+            rate = "{:.2f}".format(rate)+' B'
+        return rate+'/s'
+
+def format_prate(rate):
+    """
+    Show the packet rate.
+    """
+    if rate == STATUS_NA:
+        return STATUS_NA
+    else:
+        return "{:.2f}".format(float(rate))+'/s'
+
+def format_util(brate, port_rate=PORT_RATE):
+    """
+    Calculate the util.
+    """
+    if brate == STATUS_NA:
+        return STATUS_NA
+    else:
+        util = brate/(float(port_rate)*1024*1024*1024/8.0)*100
```

Comment:
This assumes the port speed to be Gbps. Though most of the devices have lowest speed as 1 Gbps. I don't think it's a good design here. If SONiC is planned to be used for management and other low capacity devices in the future this will become a bug for ports less than 1Gbps. I would rather prefer logic using the port speed to determine the factor to divide rather than blindly assuming Gbps

Reply:
Could you please describe the suggested approach in more detail?

Reply:
I believe port speed can be used to figure out if it is in Gbps or Mbps. By default I believe speed is in Mbps (eg 40G is 40000)